File: services/backend/plan_service/src/main/java/com/example/plan_service/presentation/http/Response.java
package com.example.plan_service.presentation.http;

public final class Response<T> {
    private final Boolean successfulOperation;
    private final Integer code;
    private final T data;
    private final String error;
    private final String message;

    public Response(Boolean successfulOperation, Integer code, T data, String error, String message){
        this.successfulOperation = successfulOperation;
        this.code = code;
        this.data = data;
        this.error = error;
        this.message = message;
    }

    public Boolean getSuccessfulOperation() {
        return successfulOperation;
    }

    public Integer getCode() {
        return code;
    }

    public T getData() {
        return data;
    }

    public String getError() {
        return error;
    }

    public String getMessage() {
        return message;
    }
}

File: services/backend/email_service/http_helper/Response.py
from typing import Any


class Response:
    __code: int
    __message: str
    __data: Any

    def __init__(self, code: int, message: str, data: Any):
        self.__code = code
        self.__data = data
        self.__message = message

    def to_dict(self) -> dict:
        return { "code": self.__code, "message": self.__message, "data": self.__data }
